### DISCUSSION TITLE
Regex Manager - Handling when version doesn't exactly match github-tags format

### DISCUSSION THREAD

**Discussion by 56KBs** [renovatebot/renovate#22187] [Suggest an Idea]
### Which Renovate are you using?

Renovate Open Source CLI

### Which platform are you using?

GitHub Enterprise

### Have you checked the logs? Don't forget to include them if relevant

[renovatelog.txt](https://github.com/renovatebot/config-help/files/4889851/renovatelog.txt)

### What would you like to do?

I'm looking to see if it is possible to use a regex manager, alongside using a regex versioning template, from testing I don't appear to be able to do what I want to do.

My configuration is:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "regexManagers": [
    {
      "fileMatch": ["\\.spec$"],
      "matchStrings": [
        "Requires:\\s+?(?<repoName>cs-[a-z-]+?)\\s+?=\\s+?(?<currentValue>(?:\\d[.]){2}\\d)"
      ],
      "depNameTemplate": "my-organisation/{{{repoName}}}",
      "versioningTemplate": "regex:^(?:v)?(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)(?:-rc\\d+?)?$",
      "datasourceTemplate": "github-tags"
    }
  ]
}
```

For example, I have a filename.spec file, which contains:
```
Requires: cs-example = 1.0.0
```

Which I have as a repository in my organisation that contains tags in github - However, they're formatted differently to this, they're formatted like so:
```
v1.0.0-rc1
v1.0.0-rc2
v1.0.1-rc1
v1.1.0-rc1
v1.1.0-rc2
```

As a result, I'm attempting to get renovate to identify the `1.0.0` version in the spec file, and match that to the github tags, replacing `1.0.0` with `1.1.0` for example.

Based on the logs I can see that is is successfully identifying that I have an update from version `1.0.0` to version `1.1.0`, but when it comes to actually doing the replacement, it appears that it uses the full github tag name, meaning replacement ultimately fails.

I'm aware that what I'm looking to do might not even actually be possible in the current form of renovate.

**Comment by rarkins** [comment]
I've transferred this into the main repo, as I don't think we can achieve this with any config. The main issue is that GitHub Tags is not a true "registry", and they can be like "1.0.0", "v1.2.0" or "v1.3.0 - Ukelele", "foo", etc. Naturally we normally focus on the first two versions and filter the rest.

One possible thing we could do is "massage" the results so that it's "1.0.0" and "1.2.0", but unfortunately this would break some use cases that *need* to have an exact match on the tag. I can think of two ways to address it:

- A whole new datasource! Called e.g. `git-tags-massaged` that simply massages off the `v` from any results
- A new boolean parameter to be passed to datasources e.g. called `massageVersion` or something similar.

In your case, the best/quickest solution would be if you are able to change `Requires: cs-example = 1.0.0` to `Requires: cs-example = v1.0.0` so that it uses the literal tag string and can replace it.
